lib/i18n-transform.js

```diff
@@ -52,93 +52,172 @@ var _getLanguageFromI18n = function(i18nObj) {
         return language;
     },
 
-    _extractPreferredLanguage = function(r, i18nArray, languages){
+    _isLanguageMatch = function (i18n, language) {
+        var i18nLanguage = _getLanguageFromI18n(i18n);
+
+        return language.code.toUpperCase() === i18nLanguage.code.toUpperCase() &&
+            ((_.isUndefined(language.region) || _.isUndefined(i18nLanguage.region)) ||
+                language.region.toUpperCase() === i18nLanguage.region.toUpperCase());
+    },
+
+    _extractPerField = function (fields, languages, i18nArray, primaryTranslations, mandatory) {
```

Comment:
- 'languages': can we call these 'acceptedLanguages'? To distinguish from the fields translation languages?
- 'mandatory': somewhere else we used the word 'required'